### PR TITLE
Fix test case 180-fd-leaks when running on Fedora

### LINKED ISSUE #3910: 180-fd-leaks.t fails on Fedora
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior
`180-fd-leaks.t` fails when running on Fedora

## Expected Behavior
`180-fd-leaks.t` passes

## Reproduction Instructions
Download i3 repository and install dependencies, below is what I explicitly installed today:
<pre>
$ sudo dnf install autoconf automake cairo-devel gcc libXinerama-devel libXrandr-devel libev-devel libxcb-devel libxkbcommon-devel libxkbcommon-x11-devel pango-devel pcre-devel perl-App-cpanminus startup-notification-devel xcb-proto xcb-util-cursor-devel xcb-util-devel xcb-util-keysyms-devel xcb-util-wm-devel xcb-util-xrm-devel xorg-x11-server-Xephyr xorg-x11-server-Xvfb yajl-devel
</pre>

Run the below commands:
<pre>
$ cd ~/dev/i3/testcases
$ sudo dnf install perl-App-cpanminus
$ sudo cpanm .
$ sudo make clean
$ cd ~/dev/i3/AnyEvent-I3
$ sudo cpanm Module::Install
$ sudo cpanm .
$ sudo make clean
$ cd ~/dev/i3
$ autoreconf -fi && mkdir build && cd build && ../configure
$ make check
</pre>

## Environment
Output of `i3 --moreversion 2>&-`:
<pre>
i3 version:  4.17.1
commit: fc01ece3657e334d9067c40c0e7274ee955345e5
</pre>

Output of `make check`:
<pre>
=================================
   i3 4.17.1: ./test-suite.log
=================================

# TOTAL: 1
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: testcases/complete-run.pl
===============================

Running tests under Xvfb display :99
Starting 32 Xephyr instances, starting at :1025...

Rough time estimate for this run: 5.78 seconds

Writing logfile to 'testsuite-2020-01-18-21-30-39-4.17-246-gfc01ece3/complete-run.log'...

#   Failed test 'No file descriptors leaked'
#   at /home/eganjs/dev/i3/build/../../i3/testcases/t/180-fd-leaks.t line 63.
#          got: '2'
#     expected: '0'

Test Summary Report
-------------------
Use of uninitialized value in subtraction (-) at /usr/share/perl5/vendor_perl/TAP/Formatter/Base.pm line 386.
Negative repeat count does nothing at /usr/share/perl5/vendor_perl/TAP/Formatter/Base.pm line 386.
/home/eganjs/dev/i3/build/../../i3/testcases/t/180-fd-leaks.t (Wstat: 0 Tests: 1 Failed: 1)
  Failed test:  1
Files=249, Tests=3588,  3 wallclock secs ( 0.50 usr +  0.05 sys =  0.55 CPU)
Result: FAIL

The slowest tests are:
	/home/eganjs/dev/i3/build/../../i3/testcases/t/308-focus_wrapping.t with 2.43 seconds
	/home/eganjs/dev/i3/build/../../i3/testcases/t/298-ipc-misbehaving-connection.t with 2.33 seconds
	/home/eganjs/dev/i3/build/../../i3/testcases/t/201-config-parser.t with 1.71 seconds
	/home/eganjs/dev/i3/build/../../i3/testcases/t/167-workspace_layout.t with 1.66 seconds
	/home/eganjs/dev/i3/build/../../i3/testcases/t/200-urgency-timer.t with 1.41 seconds
FAIL testcases/complete-run.pl (exit status: 1)
</pre>

<pre>
- Linux Distribution & Version: Fedora release 31 (Thirty One) 5.4.10-200.fc31.x86_64
- Are you using a compositor (e.g., xcompmgr or compton): no
</pre>

Output of `ll /proc/self/fd`:
<pre>
total 0
lrwx------. 1 eganjs eganjs 64 Jan 18 22:14 0 -> /dev/pts/4
lrwx------. 1 eganjs eganjs 64 Jan 18 22:14 1 -> /dev/pts/4
lrwx------. 1 eganjs eganjs 64 Jan 18 22:14 2 -> /dev/pts/4
lr-x------. 1 eganjs eganjs 64 Jan 18 22:14 3 -> /var/lib/sss/mc/passwd
lr-x------. 1 eganjs eganjs 64 Jan 18 22:14 4 -> /var/lib/sss/mc/group
lr-x------. 1 eganjs eganjs 64 Jan 18 22:14 5 -> /proc/125239/fd
</pre>

`/var/lib/sss/mc/passwd` `/var/lib/sss/mc/group` appear to be the extra file descriptors

This issue mentions that these file descriptors became part of the default configuration for Fedora at one point https://bugzilla.redhat.com/show_bug.cgi?id=1493038

In conculsion the above seems to indicate that the presence of `/var/lib/sss/mc/passwd` and `/var/lib/sss/mc/group` is breaking the test, and while I'm sure there is a way to disable them, I think it's far more usable and dev friendly to adjust the test to account for this.

### PR DESCRIPTION
**Pull request by eganjs**
Resolves #3910

### CODE REVIEW

File: testcases/t/180-fd-leaks.t [hunk 1]
```diff
@@ -57,7 +57,7 @@ delete $fds{2};
 
 # Filter out the fd which is caused by ls calling readdir().
```
Comment (xzfc) [inline testcases/t/180-fd-leaks.t]: This comment is outdated now.

**Comment by xzfc** [comment]
BTW, another possible implementation option is to run `ls /proc/self/fd` both from within i3 and perl and compare the output. In that case, no hardcoding of `/var/lib/sss/mc/...` is needed.

**Comment by eganjs** [comment]
> BTW, another possible implementation option is to run `ls /proc/self/fd` both from within i3 and perl and compare the output. In that case, no hardcoding of `/var/lib/sss/mc/...` is needed.

I think that is a more favourable approach, will update

File: testcases/t/180-fd-leaks.t (hunk 1 above)
Comment (eganjs) [reply testcases/t/180-fd-leaks.t]: Based on your suggestion, I refactored my approach, so this comment is no longer outdated

**Pull request merged**